packages/frontend/src/app/services/globals/globals.service.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Injectable} from '@angular/core';
+import {BehaviorSubject} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GlobalsService {
+  pageDataChange: BehaviorSubject<PageState> = new BehaviorSubject<PageState>({
+    currentPage: '',
+    repoName: '',
+    orgName: '',
+  });
+
+  update(pageName: string, orgName?: string, repoName?: string) {
```

Comment:
so the purpose of this global service, if I understand, is to maintain a global variable with a `repoName` and `orgName` variable.

What will `repoName` be populated with on the repo listing page, vs., the test listing page for a given repository.

Reply:
Correct that is the purpose for the global service.  On the repository page it will have the name of the repository who's tests are currently being displayed.  That is a good question regarding the organization page that lists the repos because I am not 100% sure.  Good question for @gis727 ! I believe it should have an empty value for the repository at that point since the url for the org page does not have `:repo`, but I could be wrong.

Reply:
For the "repo listing page" AKA the org page, i think it'll be empty unless the user is searching a repository (in which case it will have that repository name).
On the test listing page AKA the repo page, it'll have the repository name.